src/Console/Command/CleanMySQLDatabaseCommand.php

```diff
@@ -60,92 +57,106 @@ protected function configure(): void
             ]));
-        } catch (JiraException|\Exception $e) {
+        } catch (JiraException|Exception $e) {
             $io->error($e->getMessage());
+
             return 1;
         }
 
         $io->title('Scan databases');
 
         $connectionString = sprintf('mysql:host=%s;', $dbHost);
-        $pdo = new \PDO($connectionString, $dbUser, $dbPassword);
-        $pdo->setAttribute(\PDO::ATTR_ERRMODE, \PDO::ERRMODE_EXCEPTION);
+        $pdo              = new PDO($connectionString, $dbUser, $dbPassword);
+        $pdo->setAttribute(PDO::ATTR_ERRMODE, PDO::ERRMODE_EXCEPTION);
         $databases = array_column($pdo->query('SHOW DATABASES')->fetchAll(), 'Database');
         $databases = array_filter(
             $databases,
-            function ($dbName) use ($pattern) {
+            static function ($dbName) use ($pattern) {
                 return preg_match($pattern, $dbName);
             }
         );
 
         $remove = [];
-        $notfound = [];
 
         foreach ($databases as $database) {
 
             $branchName = !empty($branchNameFilter)
                 ? str_replace($branchNameFilter, '', $database)
                 : $database;
 
+            $issue = null;
             try {
                 $issue = $issueService->get($branchName, ['fields' => ['status']]);
+            } catch (Exception $e) {
+                if ($e->getCode() === 404) {
+                    $logger->critical(
+                        'While cleaning up, I found an unexpected database with no matching Jira Ticket.'
+                        . ' The database will be removed, but you should investigate why it was'
+                        . ' created in the first place!',
+                        [
+                            'CleanMySQLDatabaseName' => 'Error while matching databases to issues',
+                            'Unexpected database'    => $database,
+                            'Branch name'            => $branchName,
+                            'Possible cause'         => 'Jira ticket was deleted'
+                        ]
+                    );
+                } else {
+                    throw $e;
+                }
+            }
 
-                if (!empty($statuses)) {
-                    $issueStatus = strtolower($issue->fields->status->name);
-
-                    if ($invert) {
-                        if (!in_array($issueStatus, $statuses)) {
-                            $remove[] = $database;
-                            $io->text(sprintf('Found orphaned database %s', $database));
-                        }
-                    } else {
-                        if (in_array($issueStatus, $statuses)) {
-                            $remove[] = $database;
-                            $io->text(sprintf('Found orphaned database %s', $database));
-                        }
+            if (!empty($statuses)) {
+                $issueStatus = strtolower($issue->fields->status->name);
```

Comment:
This will throw a notice in case of an exception above.

Reply:
Very true. Thx!